src/Chapter4.hs

```diff
@@ -470,25 +485,42 @@ Applicatives can be found in many applications:
 
 Implement the Applicative instance for our 'Secret' data type from before.
 -}
+
 instance Applicative (Secret e) where
-    pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+  pure :: a -> Secret e a
+  pure = Reward
+
+  (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
+  (Trap e) <*> _ = Trap e
+  (Reward e) <*> f = fmap e f -- why not e f 
```

Comment:
`e f` doesn't make any sense as `e` is not a function that takes `Secret e a`, the type of `e :: a -> b`, so I can't see what do you mean by `e f` as it wouldn't even type check.